code/modules/mob/living/bot/cleanbot.dm

```diff
@@ -1,8 +1,20 @@
+// Updated by Nadrew, bits and pieces taken from Baycode, but fairly heavily modified to function here (and because a few bits of the baycode was ehh)
+
+// The main issue in the old code was the Life() loop and the fact that it could go infinite really easily.
+// The fix involved labeling the various loops involved so they could be continued and broken properly.
+// It also decreases the amount of calls to AStar() and handle_target()
+
+var/list/cleanbot_types // Going to use this to generate a list of types once then cull it out locally, see comments below for more info
+
+/obj/effect/decal/cleanable/var
+	being_cleaned = 0
+	tmp/mob/living/bot/cleanbot/clean_marked = 0 // If a cleaning bot has marked the cleanable to be cleaned, to prevent multiples from going to the same one.
```

Comment:
Ah yeah, it should probably be nulling it out before the call (it ends up set to null by other bits in the code (eventually, and hopefully), but I wouldn't count on that stuff beyond a qdel() call, so I went ahead and did it, was meant to be doing the other null stuff before qdel() and not after too, even though none of it was actually preventing it from being GC'd, it's ideal to be positive it's not going to hang around any longer than it needs to.